synthetics-stats/src/entities/userBalance.ts

```diff
@@ -0,0 +1,127 @@
+import { BigInt, log } from "@graphprotocol/graph-ts";
+import {
+  CollectedMarketFeesInfo,
+  LatestUserGmTokensBalanceChangeRef,
+  Transaction,
+  UserGmTokensBalanceChange
+} from "../../generated/schema";
+import { getOrCreateCollectedMarketFees } from "./fees";
+
+let ZERO = BigInt.fromI32(0);
+
+export function saveUserGmTokensBalanceChange(
+  account: string,
+  marketAddress: string,
+  value: BigInt,
+  transaction: Transaction,
+  transactionLogIndex: BigInt,
+  postfix: string
+): void {
+  let prevEntity = getLatestUserGmTokensBalanceChange(account, marketAddress);
+  let entity = _createUserGmTokensBalanceChange(account, marketAddress, transaction, transactionLogIndex, postfix);
+  let totalFees = CollectedMarketFeesInfo.load(marketAddress + ":total");
+  let prevBalance = prevEntity ? prevEntity.tokensBalance : ZERO;
+  let prevCumulativeIncome = prevEntity ? prevEntity.cumulativeIncome : ZERO;
+  let income = prevEntity ? calcIncomeForEntity(prevEntity) : ZERO;
```

Comment:
i think there is no need in ternary operator, `calcIncomeForEntity` handles case when entity is null